cli/spcli/sectors.go

```diff
@@ -0,0 +1,1395 @@
+func ArrayToString(array []uint64) string {
+	sort.Slice(array, func(i, j int) bool {
+		return array[i] < array[j]
+	})
+
+	var sarray []string
+	s := ""
+
+	for i, elm := range array {
+		if i == 0 {
+			s = strconv.FormatUint(elm, 10)
+			continue
+		}
+		if elm == array[i-1] {
+			continue // filter out duplicates
+		} else if elm == array[i-1]+1 {
+			s = strings.Split(s, "-")[0] + "-" + strconv.FormatUint(elm, 10)
+		} else {
+			sarray = append(sarray, s)
+			s = strconv.FormatUint(elm, 10)
+		}
+	}
+
+	if s != "" {
+		sarray = append(sarray, s)
+	}
+
+	return strings.Join(sarray, ",")
+}
+
+func SectorsCompactPartitionsCmd(getActorAddress ActorAddressGetter) *cli.Command {
+	return &cli.Command{
+		Name:  "compact-partitions",
+		Usage: "removes dead sectors from partitions and reduces the number of partitions used if possible",
+		Flags: []cli.Flag{
+			&cli.Uint64Flag{
+				Name:     "deadline",
+				Usage:    "the deadline to compact the partitions in",
+				Required: true,
+			},
+			&cli.Int64SliceFlag{
+				Name:     "partitions",
+				Usage:    "list of partitions to compact sectors in",
+				Required: true,
+			},
+			&cli.BoolFlag{
+				Name:  "really-do-it",
+				Usage: "Actually send transaction performing the action",
+				Value: false,
+			},
+			&cli.StringFlag{
+				Name:  "actor",
+				Usage: "Specify the address of the miner to run this command",
+			},
+		},
+		Action: func(cctx *cli.Context) error {
+			fullNodeAPI, acloser, err := lcli.GetFullNodeAPI(cctx)
+			if err != nil {
+				return err
+			}
+			defer acloser()
+
+			ctx := lcli.ReqContext(cctx)
+
+			maddr, err := getActorAddress(cctx)
+			if err != nil {
+				return err
+			}
+
+			minfo, err := fullNodeAPI.StateMinerInfo(ctx, maddr, types.EmptyTSK)
+			if err != nil {
+				return err
+			}
+
+			deadline := cctx.Uint64("deadline")
+			if deadline > miner.WPoStPeriodDeadlines {
+				return fmt.Errorf("deadline %d out of range", deadline)
+			}
+
+			parts := cctx.Int64Slice("partitions")
+			if len(parts) <= 0 {
+				return fmt.Errorf("must include at least one partition to compact")
+			}
+			fmt.Printf("compacting %d partitions\n", len(parts))
+
+			var makeMsgForPartitions func(partitionsBf bitfield.BitField) ([]*types.Message, error)
+			makeMsgForPartitions = func(partitionsBf bitfield.BitField) ([]*types.Message, error) {
+				params := miner.CompactPartitionsParams{
+					Deadline:   deadline,
+					Partitions: partitionsBf,
+				}
+
+				sp, aerr := actors.SerializeParams(&params)
+				if aerr != nil {
+					return nil, xerrors.Errorf("serializing params: %w", err)
+				}
+
+				msg := &types.Message{
+					From:   minfo.Worker,
+					To:     maddr,
+					Method: builtin.MethodsMiner.CompactPartitions,
+					Value:  big.Zero(),
+					Params: sp,
+				}
+
+				estimatedMsg, err := fullNodeAPI.GasEstimateMessageGas(ctx, msg, nil, types.EmptyTSK)
+				if err != nil && xerrors.Is(err, &api.ErrOutOfGas{}) {
```

Comment:
xerrors.Is was deprecated a long time ago.
```suggestion
				if err != nil && errors.Is(err, &api.ErrOutOfGas{}) {
```